internal/config/common.go

```diff
@@ -4,11 +4,14 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
-	"testing"
 
 	"github.com/BurntSushi/toml"
 )
 
+type Configurable interface {
+	Merge(config Configurable) Configurable
```

Comment:
NIT: I don't love this interface, but at least for now you should document the invariants (that seem to be the `config` should be the same type as the receiver, and this type will be returned).

Also a random nit about naming: This would seem to suggest that a configurable is an "object that can be configured", whereas it's actually an "object that represents a configuration". Maybe something like `Mergable`.

Reply:
Go is Go so there's no native way to express that invariant in the type system. I'll look into documenting it comment-wise.